src/main/java/seedu/address/logic/parser/FindCommandParser.java

```diff
@@ -30,7 +31,13 @@ public FindCommand parse(String args) throws ParseException {
             throw new ParseException(
                     String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
         }
+
+        // Verify nameKeywords are valid Name inputs
         List<String> nameKeywords = Arrays.asList(nameKeyArg.split("\\s+"));
+        if (!nameKeyArg.isEmpty() && !NameContainsKeywordsPredicate.areValidNameKeywords(nameKeywords)) {
```

Comment:
Should we handle this in NameContainsKeywordsPredicate constructor instead?

Reply:
Refactored to use ParserUtil instead to preserve SLAP